### PR TITLE
fixed the mobile responsive and percentage

### PR DESCRIPTION
**Pull request by ify47**
fixed the mobile and tablet responsiveness so users wont have an issue when using calculator, and also fixed the percentage bug, please review percentage bug if it is completely gone.

Best regards
ify

### CODE REVIEW

**Comment by N3v1** [comment]
Thank you @ify47,
If anything is not working properly, I will return to inform you.

Thank you kindly
Best regards
NH

**Comment by N3v1** [comment]
Hi @ify47 👋,
the percentage functionality still doesn't work
</a>

**Comment by N3v1** [comment]
And could you please use another script 'cause I'm getting a security warning from my code scanner "CWE-830
Inclusion of Web Functionality from an Untrusted Source" **(see #24)**

**Comment by ify47** [comment]
Oh i see, you are substracting, alright, i will get to it then, and sure, i will use another script

**Comment by N3v1** [comment]
Okay, thanks. The percent function should work with all other operators +, -, *, ...
Also, is the security warning serious? I had never received such a warning from my code scanner

Thank you kindly 
Best regards
NH

File: index.html [hunk 1]
```diff
@@ -22,6 +22,7 @@
 
     <!--Link the index.js file-->
     <script src="index.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/decimal.js/10.3.1/decimal.min.js"></script>
```
Comment (N3v1) [inline index.html:25]: Could you please change the script `cause of issue **#24 (CWC-830)**

**Comment by N3v1** [comment]
Hi @ify47, 
how is your percentage fix going? 

Best regards
NH

**Comment by N3v1** [comment]
Hi @ify47 👋,
maybe I found the issue:

```js
function percentageToDecimal(percentage) {
  return new Decimal(percentage) / 100; // Divide the percentage value by 100 to get the decimal value
}
```
```js
// Iterate through the expression parts
  splitUpCalculation.forEach((part, index) => {
    if (part === "%") {
      // If the part is a percentage symbol
      const number = splitUpCalculation[index - 1]; // Get the number before the percentage symbol
      const nC_index = newCalculation.length - 1;

      // Check if the number is already a decimal or needs conversion
      if (typeof number === "number") {
        newCalculation[nC_index] = number / 100; // Convert the number to a decimal
      } else {
        newCalculation[nC_index] = percentageToDecimal(number); // Convert the percentage value to a decimal
      }
      return;
    }

    newCalculation.push(part); // Push non-percentage parts to the newCalculation array
  });
```
```js
// Use Decimal type for calculations
  let result = new Decimal(eval(newCalculation.join(" "))); // Evaluate the new expression
  resultContainer.innerHTML = result.toString(); // Convert result to string for display
```
It seems that the problem is related to those specific lines.

I found out, that e.g. if you type 100 - 100%, you'll get 99 as a result, so the calculator thinks that 100 % of 100 is 1 and not 100. 

Bildschirmfoto 2023-08-13 um 18 16 50

---

**Example 2:** A pen costs €20.00. The tax is 19%. Calculate the price of the pen excluding the 19% tax.
20€ - 19% = 16,20€ 

| Apple Calculator | this calculator |
| :---------------: | :-------------: |
| Bildschirmfoto 2023-08-13 um 18 24 50 | Bildschirmfoto 2023-08-13 um 18 24 13 |

But if you type: `20*0.19` = 3.8 and then `20-3.8` the result is 16,20[€]

I hope this was of some assistance to you @ify47.

**Comment by ify47** [comment]
Hello my friend🖐, I'm sorry, I'm not done with it, but thanks for the assistance, i will be done soon 🙏

**Comment by jiri132** [comment]
> Hi @ify47 👋, maybe I found the issue:
> 
> ```js
> function percentageToDecimal(percentage) {
>   return new Decimal(percentage) / 100; // Divide the percentage value by 100 to get the decimal value
> }
> ```
> 
> ```js
> // Iterate through the expression parts
>   splitUpCalculation.forEach((part, index) => {
>     if (part === "%") {
>       // If the part is a percentage symbol
>       const number = splitUpCalculation[index - 1]; // Get the number before the percentage symbol
>       const nC_index = newCalculation.length - 1;
> 
>       // Check if the number is already a decimal or needs conversion
>       if (typeof number === "number") {
>         newCalculation[nC_index] = number / 100; // Convert the number to a decimal
>       } else {
>         newCalculation[nC_index] = percentageToDecimal(number); // Convert the percentage value to a decimal
>       }
>       return;
>     }
> 
>     newCalculation.push(part); // Push non-percentage parts to the newCalculation array
>   });
> ```
> 
> ```js
> // Use Decimal type for calculations
>   let result = new Decimal(eval(newCalculation.join(" "))); // Evaluate the new expression
>   resultContainer.innerHTML = result.toString(); // Convert result to string for display
> ```
> 
> It seems that the problem is related to those specific lines.
> 
> I found out, that e.g. if you type 100 - 100%, you'll get 99 as a result, so the calculator thinks that 100 % of 100 is 1 and not 100.
> 
> Bildschirmfoto 2023-08-13 um 18 16 50
> **Example 2:** A pen costs €20.00. The tax is 19%. Calculate the price of the pen excluding the 19% tax. 20€ - 19% = 16,20€
> 
> Apple Calculator	this calculator
> Bildschirmfoto 2023-08-13 um 18 24 50	Bildschirmfoto 2023-08-13 um 18 24 13
> But if you type: `20*0.19` = 3.8 and then `20-3.8` the result is 16,20[€]
> 
> I hope this was of some assistance to you @ify47.

hey @N3v1  @ify47 👋,

OH I see the thing you are trying to accomplish!

I can introduce an easy fix to this because it looks simple to be solved 

How I have created it is that it gets the decimal (20,00$ - 20%) 

Percentile: 20
Decimal: 0.2 
this is when you do multiplication 
but since you are doing the `-` calculation it does still the same as the one you need when you multiplicate.

how I would suggest it gets changed to is find the 'order' as in how BODMAS is used for example if it needs to get used in an    ` * /` or `+ -` in that way, you can find out how it needs to get calculated, this can be done by looking 2 up from the `Procentiles index`  and then if 1 down is equal to - but to the top is * then it will switch to `division by 100` else if it is only a `+ or -` 
that it would get the number below that, and do percentiles on that number 

## Example 1:
```js
const calculation = [20 , - , 50 , % , + , 50 , % ];
const desiredOutcome = 15;

function PrecetileToDecimal(Precentile) {
      let index = calculation.findIndexOf(Precentile);
      
// Check if the index is not lower then  0 when you check if there is 'sign' underneath
      if (0 < index - 1) {
           // If there is a sign underneath and it is * or / then just divide by 100 to do simple multiplication
          // else take the Prectinle make it to a nice decimal point and get the calculation form down below
           if (calculation[index - 1] === "*" || calculation[index - 1] === "/") {return Precentile/100}

           else if (calculation[index - 1] === "+" || calculation[index-1] === "-") {return Precentile / 100 * calculation[index-2] }
      }

// if there is none * or / dow there then look up if it is * or / because they have a preference when going `20 + 50% * 10` 
      if (calculation.length > index + 1) {
           if (calculation[index + 2] === "*" || calculation[index+2] === "/") {return Precentile/100}
      }

// this maybe should be just `Precentile / 100` because 50% + 50% = 0.75 then it would be like
// `0.5 + 0.25` (because the first one is just 0.5 then and the second looks down and finds a 0.5 and then it does 50/100*0.5 coming down to 0.75 after the second one)
      throw new Error("There has been an error in the function")
}
```

This is an hypothetical answer to the issue that I just thought of!

if I need to change this i will perfect this and make a better documentation tomorrow night!

kind regards,
JB

**Comment by jiri132** [comment]
And to add to this THAT PEN IS FCKING EXPENSIVE 20$ 🤣 🤣

**Pull request merged**